src/sagemaker/estimator.py

```diff
@@ -3846,8 +3848,20 @@ def _distribution_configuration(self, distribution):
             # smdistributed strategy selected
             if get_mp_parameters(distribution):
                 distribution_config["mp_parameters"] = get_mp_parameters(distribution)
+            # first make sure torch_distributed is enabled if instance type is p5
+            torch_distributed_enabled = distribution.get("torch_distributed").get("enabled", False)
             smdistributed = distribution["smdistributed"]
             smdataparallel_enabled = smdistributed.get("dataparallel", {}).get("enabled", False)
+            p5_enabled = True if self.instance_type == "p5.48xlarge" else False
```

Comment:
nit: `self.instance_type == "p5.48xlarge"` is already a `bool`
- If you really need to explicitly keep it as `bool`, use `bool(...)` i.e. the ctor.

Reply:
I actually tried this initially but got issues. Not sure if it was specific to my machine but it was blocking for no reason which is why I wrote out the explicit if condition.